examples/checkout/sdk-wallet-balance-with-nextjs/src/app/wallet-balance-with-metamask/page.tsx

```diff
@@ -0,0 +1,317 @@
+        // Get all token balances of the wallet
+        const chainId = connectedProvider._network.chainId ?? checkout.ChainId.IMTBL_ZKEVM_TESTNET;
+        const allBalancesResponse = await checkoutSDK.getAllBalances({ provider: connectedProvider, walletAddress, chainId });
+        // #enddoc get-all-balances
+        setAllBalances(allBalancesResponse);
+      }
+    };
+
+    fetchBalances();
+  }, [connectedProvider, walletAddress]);
+
+  // Ensure connectedProvider is defined before calling getTokenInfo
+  useEffect(() => {
+    const fetchTokenInfo = async () => {
+      if (connectedProvider) {
+        // #doc get-token-info
+        // Get the details of a particular token
+        const tokenAddress = "0xD61ffaece032CA6E0C469820707d677Feb4BEDD5";
+        const tokenInfo = await checkoutSDK.getTokenInfo({ provider: connectedProvider, tokenAddress });
+        // #enddoc get-token-info
+        setTokenInfo(tokenInfo);
+      }
+    };
+    fetchTokenInfo();
+  }, [connectedProvider]);
+  useEffect(() => {
+    const fetchTokenBalance = async () => {
+      if (connectedProvider && walletAddress) {
+        // #doc get-token-bal
+        // Get the balance of a particular token
+        const tokenAddress = '0xD61ffaece032CA6E0C469820707d677Feb4BEDD5'
+        const balanceResponse = await checkoutSDK.getBalance({ provider: connectedProvider, walletAddress, tokenAddress });
+        // #enddoc get-token-bal
+        setTokenBalance(balanceResponse)
+      }
+    };
+    fetchTokenBalance();
+  }, [connectedProvider, walletAddress]);
+  useEffect(() => {
+    const fetchTokenBalance = async () => {
+      if (connectedProvider && walletAddress) {
+        // #doc get-native-bal
+        // Get the balance of the native token
+        const balanceResponse = await checkoutSDK.getBalance({ provider: connectedProvider, walletAddress });
+        // #enddoc get-native-bal
+        setNativeBalance(balanceResponse)
+      }
+    };
+    fetchTokenBalance();
+  }, [connectedProvider, walletAddress]);
+  return (
+    <>
+      <Heading size="medium" className="mb-1">
+        Wallet Balance with MetaMask
+      </Heading>
+      <Button 
+        className="mb-1"
+        size="medium" 
+        onClick={async () => await connectWithMetamask()}
+        disabled={loading}
+      >
+        Connect MetaMask
+      </Button>
+        
+      <Table>
+        <Table.Head>
+          <Table.Row>
+            <Table.Cell>Item</Table.Cell>
+            <Table.Cell>Value</Table.Cell>
+          </Table.Row>
+        </Table.Head>
+        <Table.Body>
+          <Table.Row>
+            <Table.Cell><b>Supported Wallets</b></Table.Cell>
+            <Table.Cell>
+              {!supportedWallets && ' (not connected)'}
+              {supportedWallets && (
+                supportedWallets.map((wallet, index) => (
+                  <span key={index}>{wallet.walletProviderName}, </span>
+                ))
+              )}
```

Comment:
Generally you want to avoid using an AND operator in JSX, and should opt for a ternary operator for safety.

If you're interested in a little in depth on this, you can read about that [here](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx) 😄

Reply:
nice, didnt think about this. It's such a common design pattern to just use the && though. I'll swap these out for ternary now.

What do you think about the `??` [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing)? Would that work or best to stick with ternary?